tests/test_rooms/test_assign_resource.py

```diff
@@ -0,0 +1,81 @@
+from tests.base import BaseTestCase, CommonTestCases
+from fixtures.token.token_fixture import ADMIN_TOKEN
+from fixtures.room.assign_resource_fixture import (
+    assign_resource_mutation,
+    assign_resource_mutation_response,
+    assign_resource_non_existent_room,
+    assign_non_existent_resource_id,
+    assign_excess_quantity,
+    assign_quantity_less_than_one,
+    query_string
+    )
+
+import sys
+import os
+sys.path.append(os.getcwd())
+
+
+class TestAssignResource(BaseTestCase):
+
+    def test_assign_resource(self):
+        """
+            Test for assigning a resource to a room
+        """
+        CommonTestCases.admin_token_assert_equal(
+           self,
+           assign_resource_mutation,
+           assign_resource_mutation_response,
+        )
+
+    def test_assign_non_existent_resource(self):
+        """
+            Test assigning a resource that does not exist
+        """
+        CommonTestCases.admin_token_assert_in(
+            self,
+            assign_non_existent_resource_id,
+            'invalid resource id'
+        )
+
+    def test_assign_non_existent_room(self):
+        """
+            Test assigning a resource to a room that does not exist
+        """
+        CommonTestCases.admin_token_assert_in(
+            self,
+            assign_resource_non_existent_room,
+            'Room not found'
+        )
+
+    def test_assign_resource_excess_quantity(self):
+        """
+            Test assign a resource with a quantity higher than\
+            exists in database
+        """
+        CommonTestCases.admin_token_assert_in(
+            self,
+            assign_excess_quantity,
+            'assigned resource cannot exceed quantity in database.'
+        )
+
+    def test_assign_quantity_less_than_one(self):
+        """
+            Test assign a resource whose quantity is less than\
+            one
+        """
+        CommonTestCases.admin_token_assert_in(
+            self,
+            assign_quantity_less_than_one,
+            'assigned quantity cannot be less than 1.'
+        )
+
+    def test_assign_resource_multiple_times(self):
```

Comment:
Great work @d-kahara But I think this most this function's logic would have been better placed in the base.CommonTestCases then call it from here as you did with the others. What do you think?

Reply:
Also, I suppose some of the logic you are implementing here is already in the base.CommonTestCases you just need to reuse. Especially from line 75

Reply:
@TeamoreA , the test in this case hits the endpoint twice. I have not seen an implementation similar to this in the `base.CommonTestCases`. Please provide an example of the implementation you're suggesting